src/Service.Host/client/src/components/reports/OrdersBySupplierOptions.js

```diff
@@ -0,0 +1,120 @@
+        setSupplier(selectedsupplier);
+    };
+
+    const handleClick = () =>
+        history.push({
+            pathname: `/purchasing/reports/orders-by-supplier/report`,
+            search: `?id=${
+                supplier.id
+            }&fromDate=${fromDate.toISOString()}&toDate=${toDate.toISOString()}`
+        });
+
+    return (
+        <Page history={history} homeUrl={config.appRoot}>
+            <Title text="Orders By Supplier" />
+            <Grid style={{ marginTop: 40 }} container spacing={3} justifyContent="center">
+                <Grid item xs={12}>
+                    <Typography variant="h6" gutterBottom>
+                        Choose a date range:
+                    </Typography>
+                </Grid>
+                <Grid item xs={3}>
+                    <DatePicker
+                        label="From Date"
+                        value={fromDate.toString()}
+                        minDate="01/01/2000"
+                        maxDate={maxDate}
+                        onChange={newValue => {
+                            setFromDate(newValue);
+                        }}
+                    />
+                </Grid>
+                <Grid item xs={3}>
+                    <DatePicker
+                        label="To Date"
+                        value={toDate.toString()}
+                        minDate={fromDate.toString()}
+                        maxDate={maxDate}
+                        onChange={newValue => {
+                            setToDate(newValue);
+                        }}
+                    />
+                </Grid>
+                <Grid item xs={6}>
+                    <Typeahead
+                        label="Supplier"
+                        title="Search for a supplier"
+                        onSelect={handleSupplierChange}
+                        items={suppliersSearchResults}
+                        loading={suppliersSearchLoading}
+                        fetchItems={searchTerm => dispatch(suppliersActions.search(searchTerm))}
+                        clearSearch={() => dispatch(suppliersActions.clearSearch)}
+                        value={`${supplier?.id} - ${supplier?.name}`}
+                        modal
+                        links={false}
+                        debounce={1000}
+                        minimumSearchTermLength={2}
+                    />
+                </Grid>
+
+                <Grid item xs={12}>
+                    <Button
+                        color="primary"
+                        variant="contained"
+                        disabled={!fromDate && !toDate}
+                        onClick={handleClick}
+                    >
+                        Run Report
+                    </Button>
+                </Grid>
+            </Grid>
+        </Page>
+    );
+}
+
+OrdersBySupplierReportOptions.propTypes = {
+    history: PropTypes.shape({ push: PropTypes.func }).isRequired
```

Comment:
history isn't passed in as a prop so I imagine this PropTypes is doing nada

Reply:
Yeah will we ever have props now? Do we ditch propTypes completely?

Reply:
We'll still have to set PropTypes for all Objects we new up such a subcomponents etc. We just don't have as many now because we don't always have the "child" component to the container each time